src/megaapi_impl.cpp

```diff
@@ -9950,9 +9950,9 @@ void MegaApiImpl::getUserAlias(MegaHandle uh, MegaRequestListener *listener)
 void MegaApiImpl::setUserAlias(MegaHandle uh, const char *alias, MegaRequestListener *listener)
 {
     MegaRequestPrivate *request = new MegaRequestPrivate(MegaRequest::TYPE_SET_ATTR_USER, listener);
-    MegaStringMap stringMap;
-    stringMap.set(Base64Str<MegaClient::USERHANDLE>(uh), alias ? alias : "");
-    request->setMegaStringMap(&stringMap);
+    unique_ptr<MegaStringMap> stringMap(new MegaStringMapPrivate());
```

Comment:
Good catch!

Actually, the old declaration of the `stringMap` variable should result in a compilation error. We should define an default ctor as protected, to avoid this kind of issues. Please, add it to every public object that provides a `createInstance()` method, since those are the risky ones (for the other types, it's always the SDK who creates the private object and provides it to the app via a pointer to the public object).

Then, the right declaration here would become:
```suggestion
    unique_ptr<MegaStringMap> stringMap(MegaStringMap::createInstance());
```